lib/faraday.ml

```diff
@@ -51,7 +49,7 @@ module Deque(T:sig type t val sentinel : t end) : sig
   val is_empty : t -> bool
 
   val enqueue : elem -> t -> unit
-  val dequeue_exn : t -> elem
+  val dequeue: t -> elem option
```

Comment:
Using options will cause allocation, which is what the use of exceptions was avoiding.